public/app/plugins/panel/barchart/utils.ts

```diff
@@ -342,13 +343,17 @@ export function prepareBarChartDisplayValues(
     if (!xField) {
       return { warn: 'Configured x field not found' };
     }
+
+    if (xField.type === FieldType.number) {
+      xField = convertFieldType(xField, { destinationType: FieldType.string });
+    }
```

Comment:
I'm not sure how this helps? @leeoniya, is that some internal uPlot requirement?